packages/databyss-ui/components/PageContent/ArchiveBin.js

```diff
@@ -30,16 +35,45 @@ export const ArchiveBin = ({ pages }) => {
     setTimeout(() => archivePage(id), 50)
   }
 
+  const menuItems = [
+    {
+      icon: <ArchiveSvg />,
+      label: 'Archive',
+      shortcut: 'Ctrl + Del',
+    },
+  ]
+
   return canBeArchived ? (
-    <BaseControl
-      onClick={onClick}
-      hoverColor="background.2"
-      p="tiny"
-      title="Archive Page"
-    >
-      <Icon sizeVariant="small" color="text.3">
-        <TrashSvg />
-      </Icon>
-    </BaseControl>
+    <View position="relative">
+      <BaseControl
+        onPress={() => setShowMenu(!showMenu)}
+        hoverColor="background.2"
+        p="tiny"
+        label="Archive Page"
+      >
+        <Icon sizeVariant="medium" color="text.1">
+          <MenuSvg />
+        </Icon>
+      </BaseControl>
+      {showMenu && (
+        <ClickAwayListener onClickAway={() => setShowMenu(false)}>
+          <DropdownContainer
+            open={showMenu}
+            position={{
+              top: 36,
```

Comment:
This won't scale relative to the size of the Icon/BaseControl. Can we make a constant or theme value for `menuLauncherSize`? I think this should be the same as `editorMarginMenuItemHeight`, so maybe we can just rename that one to be more generic...